### ISSUE TITLE
Wrong #inspect is used in Method#inspect

### ISSUE THREAD

**Issue by apotonick** [jruby/jruby#4995]
### Environment

* JRuby version `jruby 9.1.7.0 (2.3.1) 2017-01-11 68056ae OpenJDK 64-Bit Server VM 25.151-b12 on 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12 +jit [linux-x86_64]`
* Ubuntu 16.04

### Expected Behavior

I override `Module.inspect` via a module (`Inspect`), `A.inspect` produces the correct result.

```ruby
module Inspect
  def inspect
    "nice view!"
  end
end

module A
  extend Inspect

  def self.a
  end
end

A.inspect #=> "nice view!"
```

I am expecting the same `inspect` being used when inspecting a method instance from `A`.

```ruby
puts A.method(:a).inspect #=> #<Method: nice view!.a>
```
This is the behavior in Ruby 2.5.

### Actual Behavior

However, in Jruby it seems as if `inspect` doesn't use the correct version of `A.inspect`, but the original one.

```ruby
# jruby 9.1.7
puts A.method(:a).inspect #=> #<Method: A.a>
```

**Comment by headius** [comment]
I have pushed a branch with fixes that re-align our Method#inspect logic with CRuby. Once it clears CI I'll merge.

**Comment by headius** [comment]
@apotonick Can you put together a spec for ruby/spec?

**Comment by apotonick** [comment]
Sure, @headius! Thanks for the fix!

In which directory would that go in ruby/spec? `core/module` ?

**Comment by headius** [comment]
@apotonick core/method/inspect_spec.rb should do it.

Fix merged to 9.1 and master!

**Comment by headius** [comment]
@apotonick Still having trouble confirming the fix? I just ran it against 9.1.16 snapshot and it does indeed seem to be fixed:

```
$ jruby -v blah.rb
jruby 9.1.16.0-SNAPSHOT (2.3.3) 2018-02-01 74d4306 Java HotSpot(TM) 64-Bit Server VM 25.131-b11 on 1.8.0_131-b11 +jit [darwin-x86_64]
"nice view!"
"#<Method: nice view!.a>"
```

**Comment by apotonick** [comment]
Sorry, I was being an idiot and still had my `if Ruby == Jruby` tests in place to make the wrong behavior pass. All good now, thanks!
